packages/java/engine-runtime/src/main/java/com/vaadin/hilla/internal/AbstractTaskEndpointGenerator.java

```diff
@@ -69,26 +73,45 @@ protected void prepareEngineConfiguration()
         if (firstRun) {
             logger.debug("Configure Hilla engine using build system plugin");
 
-            try {
-                // Create a runner for Maven
-                MavenRunner
-                        .forProject(projectDirectory, "-q", "vaadin:configure")
-                        // Create a runner for Gradle. Even if Gradle is not
-                        // supported yet, this is useful to emit an error
-                        // message if pom.xml is not found and build.gradle is
-                        .or(() -> GradleRunner.forProject(projectDirectory,
-                                "-q", "hillaConfigure"))
-                        // If no runner is found, throw an exception.
-                        .orElseThrow(() -> new IllegalStateException(String
-                                .format("Failed to determine project directory for dev mode. "
-                                        + "Directory '%s' does not look like a Maven or "
-                                        + "Gradle project.", projectDirectory)))
-                        // Run the first valid runner
-                        .run(null);
-                firstRun = false;
-            } catch (CommandRunnerException e) {
+            var mavenConfigure = MavenRunner.forProject(projectDirectory, "-q",
+                    "vaadin:configure");
+            var mavenConfigureVersion = Platform.getVaadinVersion()
+                    .flatMap(version -> MavenRunner.forProject(projectDirectory,
+                            "-q", "com.vaadin:vaadin-maven-plugin:" + version));
```

Comment:
It looks like the goal is missing here
```suggestion
                            "-q", "com.vaadin:vaadin-maven-plugin:" + version+":configure"));
```